packages/viewport/src/with-viewport-match.js

```diff
@@ -33,11 +103,33 @@ import { withSelect } from '@wordpress/data';
  * @return {Function} Higher-order component.
  */
 const withViewportMatch = ( queries ) => createHigherOrderComponent(
-	withSelect( ( select ) => {
-		return mapValues( queries, ( query ) => {
-			return select( 'core/viewport' ).isViewportMatch( query );
-		} );
-	} ),
+	( WrappedComponent ) => {
+		return ( props ) => {
+			const globalQueriesResult = useSelect( ( select ) => {
+				return mapValues( queries, ( query ) => {
+					return select( 'core/viewport' ).isViewportMatch( query );
+				} );
+			}, [ queries ] );
+			const localQueries = useContext( ViewportMatchWidthContext );
+			const localQueriesResult = useMemo(
+				() => {
+					if ( ! localQueries ) {
```

Comment:
Some comments here to explain the logic would be useful to help clarify.

```suggestion
					// Only evaluate local queries if within a width context provider.
					if ( ! localQueries ) {
```